agents/utils/network.py

```diff
@@ -1,11 +1,24 @@
 """Utilities for interacting with the Game Engine over the network."""
 
 import socketio
-import json, time
+import json
 
+def unimplemented_response(event_type : str):
+    def raiser(options):
+        raise NotImplementedError("Responding to " + event_type + " not implemented")
+    return raiser
+
+def unimplemented_update():
+    def do_nothing(event):
+        pass
+    return do_nothing
+
+def start(on_action=unimplemented_response("actions"), 
+          on_reaction=unimplemented_response("reactions"), 
+          on_card=unimplemented_response("card selection"), 
+          on_exchange=unimplemented_response("exchanges"),
+          update_f=unimplemented_update()):
```

Comment:
All of this is in the Agent class now - why was it added back?

Reply:
I think it is cause I didn't have the changes